### ISSUE TITLE
Question about the Use of Pretrained Models

### ISSUE THREAD

**Issue by fairyosme** [david-gimeno/tailored-avsr#1]
Thank you for sharing the model checkpoints in Zenodo.

I'm curious if these models need pre - training before reaching the reported results, or can be used directly?

Thanks!

**Comment by david-gimeno** [comment]
Hi @fairyosme ,

Thanks for your interest in our work :) The pre-trained weights are ready to be used directly and get the predictions and performance reported in our paper. Note, however, that slight differences in the preprocessing of the datasets might result in slight differences in terms of model performance, but they shouldn't be significant. You can find an example of an inference run in the README of the repo. But please, feel free to ask any questions.

Another important detail is that you can further fine-tune our models to your dataset of interest using our training scripts.

Best.

P.S. We are currently building a more noise-robust variant of our models. As soon as they are ready, we will also upload the checkpoints :)

**Comment by fairyosme** [comment]
Thank you for your response! I will carefully go through the inference details and make sure everything is clear. If I have any questions, I will reach out.

Thanks again!

**Comment by fairyosme** [comment]
> Hi @fairyosme ,
> 
> Thanks for your interest in our work :) The pre-trained weights are ready to be used directly and get the predictions and performance reported in our paper. Note, however, that slight differences in the preprocessing of the datasets might result in slight differences in terms of model performance, but they shouldn't be significant. You can find an example of an inference run in the README of the repo. But please, feel free to ask any questions.
> 
> Another important detail is that you can further fine-tune our models to your dataset of interest using our training scripts.
> 
> Best.
> 
> P.S. We are currently building a more noise-robust variant of our models. As soon as they are ready, we will also upload the checkpoints :)

Hi,

I recently re-downloaded the code and models you released, and tested the AVSR and ASR models on the LRS2 dataset. However, after loading the models, the WER is still quite high. Do you have any suggestions for potential solutions?

Thank you!

**Comment by david-gimeno** [comment]
Hi,

Can you share with me the command you used, pls? I run the ASR inference for LRS2 and, after fixing some typos in the configuration file (i will update the repo later), I got the expected 3.9 %WER.

**Comment by fairyosme** [comment]
> Hi,
> 
> Can you share with me the command you used, pls? I run the ASR inference for LRS2 and, after fixing some typos in the configuration file (i will update the repo later), I got the expected 3.9 %WER.

Thank you for your prompt reply. Here are the commands I have executed:
1、Cloned the Repository:
Downloaded the project from GitHub using git clone.
2、Downloaded and placed the  configuration and checkpoint files from the Model Zoo:
I obtained the configuration files and model from the Model Zoo.
3、Handled LRS2 Dataset:
Downloaded the LRS2 dataset files，Configured the split file to ensure that each audio file corresponds correctly to its respective language label.
4、Updated Configuration:
Added `task: asr` to the configuration file (asr_tailored_english.yaml) to specify the task.
Changed `encoder: my_e_branchformer` to `encoder: branchformer`  in the configuration file to update the encoder model.
Running the Project:
I used the following parameters to execute the project:
```
--config-file ./configs/ASR/asr_tailored_english.yaml
--output-dir ./asr/1-13
--output-name lrs2
--load-checkpoint ./asr/asr_tailored_english.pth
```
After performing these steps, I ran the project and observed that the Word Error Rate (WER) was approximately the same as the results obtained without loading the model. This suggests that the model might not be loading correctly or not influencing the results as expected.
I would greatly appreciate any further guidance or suggestions to resolve this issue

**Comment by david-gimeno** [comment]
I am seeing some issues with the `.yaml` of the zip file I uploaded to zenodo. They correspond to an older version of the code. I am sorry and I would update them. Anyway, this is not the reason. I am observing other issues with your command that would explain why you are not getting the expected performance.

The good point is that in the repo we already have the folder `configs/`, which includes similar configuration files. So we can use one of these configurations in our command. Another detail is that it seems you didn't specify `--mode inference` and you didn't include the LM in the inference beam search. So, run this command please:

```
python avsr_main.py \
    --test-dataset [PATH_TEST_CSV] \
    --mode inference \
    --config-file ./model_checkpoints/asr/asr_tailored_english.yaml \
    --load-checkpoint ./model_checkpoints/asr/asr_tailored_english.pth \
    --lm-config-file ./model_checkpoints/lm/lm_english.yaml \
    --load-lm ./model_checkpoints/lm/lm_english.pth \ 
    --output-dir ./asr/tailored/ --output-name lrs2
```

With this command I got around 4 %WER, which is the result reported in our paper. If you are not getting this result... isnt there any error when running the script? Please check therefore your data.

**Comment by david-gimeno** [comment]
After a long discussion, some repository and model checkpoint updates to fix small bugs, and inspecting the audio waveforms, we found the solution to the issue :) !! Please, as a preprocessing for audio cues, you should run this ffmpeg command:

```
ffmpeg -y -i ${INPUT_WAV_PATH} -ar 16000 -ac 1 -acodec pcm_s16le ${OUTPUT_WAV_PATH}
```
in all your audio signals. This results in mono-channel 16kHZ audio waveforms. The type of signals our models were trained with.
